packages/fluentui/react/src/themes/teams/components/Checkbox/checkboxStyles.ts

```diff
@@ -80,53 +116,61 @@ const checkboxStyles: ComponentSlotStylesPrepared<CheckboxStylesProps, CheckboxV
     ...(p.disabled &&
       p.checked && {
         color: v.disabledCheckedIndicatorColor,
-        background: v.disabledBackgroundChecked,
-        borderColor: 'transparent'
+        borderColor: v.disabledBackgroundChecked,
+        backgroundImage: checkboxIndicatorUrl(v.disabledCheckedIndicatorColor, v.disabledBackgroundChecked)
       })
   }),
 
   toggle: ({ props: p, variables: v }): ICSSInJSStyle => ({
     '-ms-grid-row-align': 'center',
     gridColumn: p.labelPosition === 'start' ? 3 : 1,
     boxShadow: 'unset',
+    boxSizing: 'border-box',
 
     background: v.background,
     borderColor: v.borderColor,
     borderStyle: v.borderStyle,
     borderRadius: v.toggleBorderRadius,
     borderWidth: v.borderWidth,
-    color: v.borderColor,
     margin: v.toggleMargin,
-    padding: v.togglePadding,
-    transition: 'padding .3s ease',
     userSelect: 'none',
     width: v.toggleWidth,
     height: v.toggleHeight,
 
-    [`& svg`]: {
-      width: v.toggleIndicatorSize,
-      height: v.toggleIndicatorSize
+    ':before': {
+      ...commonToggleBeforeStyles(v),
+      borderColor: p.disabled ? v.disabledToggleIndicatorColor : v.borderColor,
+      borderStyle: v.borderStyle,
+      borderWidth: v.borderWidth,
+      margin: v.togglePadding
     },
 
-    ...(p.checked && {
-      background: v.checkedBackground,
-      borderColor: v.checkedBorderColor,
-      color: v.checkedIndicatorColor,
-      padding: v.toggleCheckedPadding
-    }),
+    ...(p.checked &&
+      !p.disabled && {
```

Comment:
https://github.com/OfficeDev/office-ui-fabric-react/pull/12183/files#r388786483

Same thing about `!p.disabled`... We have following states:
- default
- `checked`
- `disabled`
- `checked` & `disabled`

Why we need negative conditions for other states here?